src/managers/backup.py

```diff
@@ -176,6 +179,50 @@ def format_backups_table(
 
         return out_f.getvalue()
 
+    def is_snapshot_in_bucket(self, backup_id: str) -> bool:
+        """Check whether the requested snapshot to restore is in the object storage."""
+        try:
+            content = self.bucket.meta.client.head_object(
+                Bucket=self.bucket.name, Key=os.path.join(self.backups_path, backup_id, "snapshot")
+            )
+        except ClientError as ex:
+            if "(404)" in ex.args[0]:
+                return False
+            raise
+        return bool(content.get("ResponseMetadata", None))
+
+    def restore_snapshot(self, backup_id: str, workload: ZKWorkload) -> None:
```

Comment:
**question:** Why prepend `bash` on the execs here instead of just running the commands directly? I don't think we need to specify it for anything in particular no?